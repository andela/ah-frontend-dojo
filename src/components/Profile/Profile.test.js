import React from 'react';
import { shallow, mount } from 'enzyme';
import ProfileMain from 'components/Profile/Index';

describe('Setting Component', () => {
  const props = {
    match:
    {
      params: {
        profileUser: 'nadralia',
      }
    },
    authenticatedUser: 'nadralia',
  };

  it('should match the snapshot', () => {
    const component = shallow(<ProfileMain {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const wrapper = mount(<ProfileMain {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
