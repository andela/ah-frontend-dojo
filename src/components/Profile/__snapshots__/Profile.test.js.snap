// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting Component should match the snapshot 1`] = `
<div
  className="main"
>
  <div
    className="profile-tab-menu"
  >
    <Tab
      className="tab"
      grid={
        Object {
          "paneWidth": 12,
          "tabWidth": 4,
        }
      }
      menu={
        Object {
          "pointing": true,
          "secondary": true,
        }
      }
      panes={
        Array [
          Object {
            "menuItem": "Articles ",
            "render": [Function],
          },
          Object {
            "menuItem": "| Settings",
            "render": [Function],
          },
        ]
      }
      renderActiveOnly={true}
    />
  </div>
</div>
`;

exports[`Setting Component should render correctly 1`] = `
<ProfileMain
  authenticatedUser="nadralia"
  match={
    Object {
      "params": Object {
        "profileUser": "nadralia",
      },
    }
  }
>
  <div
    className="main"
  >
    <div
      className="profile-tab-menu"
    >
      <Tab
        className="tab"
        grid={
          Object {
            "paneWidth": 12,
            "tabWidth": 4,
          }
        }
        menu={
          Object {
            "pointing": true,
            "secondary": true,
          }
        }
        panes={
          Array [
            Object {
              "menuItem": "Articles ",
              "render": [Function],
            },
            Object {
              "menuItem": "| Settings",
              "render": [Function],
            },
          ]
        }
        renderActiveOnly={true}
      >
        <div
          className="tab"
        >
          <Menu
            activeIndex={0}
            items={
              Array [
                "Articles ",
                "| Settings",
              ]
            }
            onItemClick={[Function]}
            pointing={true}
            secondary={true}
          >
            <div
              className="ui pointing secondary menu"
            >
              <MenuItem
                active={true}
                content="Articles "
                index={0}
                key="Articles "
                name="Articles "
                onClick={[Function]}
              >
                <a
                  className="active item"
                  onClick={[Function]}
                >
                  Articles 
                </a>
              </MenuItem>
              <MenuItem
                active={false}
                content="| Settings"
                index={1}
                key="| Settings"
                name="| Settings"
                onClick={[Function]}
              >
                <a
                  className="item"
                  onClick={[Function]}
                >
                  | Settings
                </a>
              </MenuItem>
            </div>
          </Menu>
          <TabPane
            active={true}
            as={[Function]}
            attached={true}
          >
            <Segment
              attached={true}
              className="active tab"
            >
              <div
                className="ui attached segment active tab"
              />
            </Segment>
          </TabPane>
        </div>
      </Tab>
    </div>
  </div>
</ProfileMain>
`;
